src/renderer/components/Visualiser/store/actions.js

```diff
@@ -260,26 +269,33 @@ export default {
           }
         });
 
-        (await Promise.all(explanationPromises)).map(expl => expl.getAnswers()).reduce(collect, []).forEach((expl) => {
-          explanationResult.push(expl);
-        });
+        const explanations = await Promise.all(explanationPromises);
+        const explanationsWhens = await Promise.all(explanations.map(explanation => explanation.getRule().getWhen()));
 
-        if (explanationResult.length > 0) {
-          const data = await CDB.buildInstances(explanationResult);
-          const rpData = await CDB.buildRPInstances(explanationResult, data, false, graknTx);
-          data.nodes.push(...rpData.nodes);
-          data.edges.push(...rpData.edges);
-
-          state.visFacade.addToCanvas(data);
-          commit('updateCanvasData');
-          const nodesWithAttributes = await computeAttributes(data.nodes, graknTx);
-
-          state.visFacade.updateNode(nodesWithAttributes);
-          const styledEdges = data.edges.map(edge => ({ ...edge, label: edge.hiddenLabel, ...state.visStyle.computeExplanationEdgeStyle() }));
-          state.visFacade.updateEdge(styledEdges);
-          commit('loadingQuery', false);
+        if (explanationsWhens.some(when => isRelUnassigned(when))) {
```

Comment:
I think it would be super useful if we could also return the rule name that is missing the relation vars

Reply:
was this resolved?

Reply:
the rewrite of the explanation retrieval (including this change as well as this one: https://github.com/graknlabs/workbase/pull/303#discussion_r444805436) were resolved in https://github.com/graknlabs/workbase/pull/303/commits/5e22184ce347bb7f813a10be3254fd81e462bba8